src/utils/staking.js

```diff
@@ -0,0 +1,278 @@
+import * as nearApiJs from 'near-api-js'
+import sha256 from 'js-sha256';
+import BN from 'bn.js'
+
+const {
+    transactions: {
+        functionCall
+    },
+    utils: {
+        format: {
+            parseNearAmount
+        }
+    }
+} =  nearApiJs
+
+// 125 - 175 Tgas for lockup methods
+const GAS_STAKE = '175000000000000'
+
+// custom lockups for testing (self deployed)
+const testLockup = {
+    'multisig.testnet': 'lu3.testnet',
+    'xa4.testnet': 'lu3.testnet',
+    'patrick12.testnet': 'patrick13.patrick12.testnet'
+}
+
+const stakingMethods = {
+	viewMethods: [
+		'get_account_staked_balance',
+		'get_account_unstaked_balance',
+		'get_account_total_balance',
+		'is_account_unstaked_balance_available',
+		'get_total_staked_balance',
+		'get_owner_id',
+		'get_reward_fee_fraction',
+	],
+	changeMethods: [
+		'ping',
+        'deposit',
+        'deposit_and_stake',
+        'deposit_to_staking_pool',
+		'stake',
+		'unstake',
+		'withdraw',
+	],
+}
+
+const lockupMethods = {
+    viewMethods: [
+        'get_balance',
+        'get_owners_balance',
+        'get_staking_pool_account_id',
+        'get_known_deposited_balance',
+    ]
+}
+
+const BETANET_WHITELIST = [
+    'chorus-one-pool-v1.stakehouse.betanet',
+    'crypto-solutions.stakehouse.betanet',
+    'delphidigital.stakehouse.betanet',
+    'genesislab.stakehouse.betanet',
+]
+
+const getLockupId = async (accountId) => {
+    const hash = new Uint8Array(sha256.sha256.array(accountId));
+    return Buffer.from(hash).toString('hex').substr(0, 40) + '.lockup.near'
```

Comment:
`.lockup.near` shouldn't be hardcoded. There is existing logic in wallet to do this, best to reuse the code as is